pipeline_tools/create_envelope.py

```diff
@@ -4,56 +4,94 @@
 import json
 import argparse
 from .dcp_utils import get_auth_token, make_auth_header
+from tenacity import retry, wait_exponential, stop_after_delay
 
+RETRY_SECONDS = 10
+TIMEOUT_SECONDS = 600
 
-def run(submit_url, analysis_json_path, schema_version):
+
+def run(submit_url, analysis_json_path, schema_version, retry_seconds, timeout_seconds):
     # 0. Get Auth token and make auth headers
     print('Fetching auth token from Auth0')
     auth_token = get_auth_token()
     print('Making auth headers')
     auth_headers = make_auth_header(auth_token)
 
     # 1. Get envelope url
+    envelope_url = get_envelope_url.retry_with(wait=wait_exponential(retry_seconds),
+                                               stop=stop_after_delay(timeout_seconds))(submit_url, auth_headers)
+
+    # 2. Create envelope
+    envelope_js = create_submission_envelope.retry_with(wait=wait_exponential(retry_seconds),
+                                                        stop=stop_after_delay(timeout_seconds))(envelope_url, auth_headers)
+
+    submission_url = get_entity_url(envelope_js, 'submissionEnvelope')
+    with open('submission_url.txt', 'w') as f:
+        f.write(submission_url)
+
+    # 3. Create analysis
+    with open(analysis_json_path) as f:
+        analysis_json_contents = json.load(f)
+    analyses_url = get_entity_url(envelope_js, 'processes')
+    analysis_js = create_analysis.retry_with(wait=wait_exponential(retry_seconds),
+                                             stop=stop_after_delay(timeout_seconds))(analyses_url, auth_headers, analysis_json_contents)
+
+    # 4. Add input bundles
+    input_bundles_url = get_entity_url(analysis_js, 'add-input-bundles')
+    add_input_bundles.retry_with(wait=wait_exponential(retry_seconds),
+                                 stop=stop_after_delay(timeout_seconds))(input_bundles_url, auth_headers, analysis_json_contents)
+
+    # 5. Add file references
+    file_refs_url = get_entity_url(analysis_js, 'add-file-reference')
+    print('Adding file references at {0}'.format(file_refs_url))
```

Comment:
nit: These prints are moved from legacy code, but it would be really great if we could use logging module here! To make loggers output to both logs and stdouts, there is an example of using 2 handlers: https://stackoverflow.com/questions/14058453/making-python-loggers-output-all-messages-to-stdout-in-addition-to-log 

If there is too much to do here, we can make it separate from this PR tho.

Reply:
Is it okay to make this change in a separate PR? I agree logging would be a great improvement, but this PR is already pretty big!

Reply:
Sure! I agree to separate it in a separate PR, and also, it is a "Should-have" instead of a "Mut have" item.